### PR TITLE
Add e2e for changing controller params.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #2116

### Commit Message / Changelog Entry

```bash
chore: add e2e test for changing controller params.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/interchain_accounts/params_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	chainA, _ := s.GetChains()
+	chainAVersion := chainA.Config().Images[0].Version
+
+	// setup 2 accounts: controller account on chain A, a second chain B account.
+	// host account will be created when the ICA is registered
+	controllerAccount := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	controllerAddress := controllerAccount.FormattedAddress()
+
+	// Assert that default value for enabled is true.
+	t.Run("validate the controller is enabled by default", func(t *testing.T) {
```
Comment (crodriguezvega) [inline e2e/tests/interchain_accounts/params_test.go:53]:
```suggestion
	t.Run("ensure controller is enabled", func(t *testing.T) {
```

File: e2e/tests/interchain_accounts/params_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+// QueryControllerParams queries the params for the controller
+func (s *InterchainAccountsParamsTestSuite) QueryControllerParams(ctx context.Context, chain ibc.Chain) controllertypes.Params {
+	queryClient := s.GetChainGRCPClients(chain).ICAControllerQueryClient
+	res, err := queryClient.Params(ctx, &controllertypes.QueryParamsRequest{})
+	s.Require().NoError(err)
+
+	return *res.Params
+}
+
+func (s *InterchainAccountsParamsTestSuite) TestControllerParams() {
```
Comment (crodriguezvega) [inline e2e/tests/interchain_accounts/params_test.go:37]: To make the function naming similar to the params tests we have for transfer, for example.

```suggestion
func (s *InterchainAccountsParamsTestSuite) TestControllerEnabledParam() {
```

File: e2e/tests/interchain_accounts/params_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			changes := []paramsproposaltypes.ParamChange{
+				paramsproposaltypes.NewParamChange(controllertypes.StoreKey, string(controllertypes.KeyControllerEnabled), "false"),
+			}
+
+			proposal := paramsproposaltypes.NewParameterChangeProposal(ibctesting.Title, ibctesting.Description, changes)
+			s.ExecuteGovProposal(ctx, chainA, controllerAccount, proposal)
+		}
+	})
+
+	t.Run("validate the param was successfully changed", func(t *testing.T) {
```
Comment (crodriguezvega) [inline e2e/tests/interchain_accounts/params_test.go:79]:
```suggestion
	t.Run("validate controller is disabled", func(t *testing.T) {
```

File: e2e/tests/interchain_accounts/params_test.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+// QueryControllerParams queries the params for the controller
+func (s *InterchainAccountsParamsTestSuite) QueryControllerParams(ctx context.Context, chain ibc.Chain) controllertypes.Params {
+	queryClient := s.GetChainGRCPClients(chain).ICAControllerQueryClient
+	res, err := queryClient.Params(ctx, &controllertypes.QueryParamsRequest{})
+	s.Require().NoError(err)
+
+	return *res.Params
+}
+
```
Comment (crodriguezvega) [inline e2e/tests/interchain_accounts/params_test.go:36]: Nit: add godoc.

File: e2e/tests/interchain_accounts/params_test.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	t := s.T()
+	ctx := context.TODO()
+
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	_, _ = s.SetupChainsRelayerAndChannel(ctx)
+	chainA, _ := s.GetChains()
+	chainAVersion := chainA.Config().Images[0].Version
+
+	// setup 2 accounts: controller account on chain A, a second chain B account.
```
Comment (crodriguezvega) [inline e2e/tests/interchain_accounts/params_test.go:47]:
```suggestion
	// setup controller account on chain A
```

**Review by crodriguezvega — COMMENTED** [review]
Thank for adding this, @DimitrisJim. I left a bunch of minor suggestions.

And a question: should we add this test to the appropriate .json files for compatibility testing?

**Comment by DimitrisJim** [comment]
> And a question: should we add this test to the appropriate .json files for compatibility testing?

I've added this as a point to the epic in #2010. IIRC this has only been done for client so far so we can add these all together in a single PR.

File: e2e/tests/interchain_accounts/params_test.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	paramsproposaltypes "github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+
+	controllertypes "github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/controller/types"
+	icatypes "github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/types"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/stretchr/testify/suite"
```
Comment (colin-axner) [inline e2e/tests/interchain_accounts/params_test.go:17]: a bit unconventional import structing. Usually its 
```
// standard lib

// external imports

// ibc-go imports
```

but we haven't established a practice for individual go.mod's still within the ibc-go repo

File: e2e/tests/interchain_accounts/params_test.go [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+)
+
+func TestInterchainAccountsParamsTestSuite(t *testing.T) {
```
Comment (colin-axner) [inline e2e/tests/interchain_accounts/params_test.go:20]: I guess we will need to add this to our compatibility test matrix

**Review by colin-axner — APPROVED** [review]
Terrific work my friend!

File: e2e/tests/interchain_accounts/params_test.go (hunk 6 above)
Comment (DimitrisJim) [reply e2e/tests/interchain_accounts/params_test.go:17]: yeah I'll fix this up. Ideally we should add the same linting targets in the e2e makefile to have all go code follow a similar structure.

**Review by crodriguezvega — APPROVED** [review]
Dope! 😄

**Pull request merged**